container/container.go

```diff
@@ -125,13 +125,19 @@ func GetLogs(client DockerClient, containerId string, since int64, stdout io.Wri
 func ConfigForTask(taskInfo *mesos.TaskInfo, forceCpuLimit bool, forceMemoryLimit bool, envVars []string) *docker.CreateContainerOptions {
 	labels := LabelsForTask(taskInfo)
 
+	var command []string
+	if taskInfo.Command != nil {
+		command = taskInfo.Command.Arguments
+	}
+
 	config := &docker.CreateContainerOptions{
 		Name: GetContainerName(taskInfo.TaskId),
 		Config: &docker.Config{
 			Env:          EnvForTask(taskInfo, labels, envVars),
 			ExposedPorts: PortsForTask(taskInfo),
 			Image:        *taskInfo.Container.Docker.Image,
 			Labels:       labels,
+			Cmd:          command,
```

Comment:
I'm pretty sure you can't pass this in without breaking things if you want it to run the default. I think even an empty slice will override the command from the container image. Not 100% sure about that.

Reply:
Thanks for the review @relistan, that does seems plausible and risky as is. Would this be a better approach or is there another pattern to use here instead?
```
diff --git a/container/container.go b/container/container.go
index 7645624..6b16ebc 100644
--- a/container/container.go
+++ b/container/container.go
@@ -125,20 +125,20 @@ func GetLogs(client DockerClient, containerId string, since int64, stdout io.Wri
 func ConfigForTask(taskInfo *mesos.TaskInfo, forceCpuLimit bool, forceMemoryLimit bool, envVars []string) *docker.CreateContainerOptions {
 	labels := LabelsForTask(taskInfo)
 
-	var command []string
+	dockerConfig := &docker.Config{
+		Env:          EnvForTask(taskInfo, labels, envVars),
+		ExposedPorts: PortsForTask(taskInfo),
+		Image:        *taskInfo.Container.Docker.Image,
+		Labels:       labels,
+	}
+
 	if taskInfo.Command != nil {
-		command = taskInfo.Command.Arguments
+		dockerConfig.Cmd = taskInfo.Command.Arguments
 	}
 
 	config := &docker.CreateContainerOptions{
-		Name: GetContainerName(taskInfo.TaskId),
-		Config: &docker.Config{
-			Env:          EnvForTask(taskInfo, labels, envVars),
-			ExposedPorts: PortsForTask(taskInfo),
-			Image:        *taskInfo.Container.Docker.Image,
-			Labels:       labels,
-			Cmd:          command,
-		},
+		Name:   GetContainerName(taskInfo.TaskId),
+		Config: dockerConfig,
 		HostConfig: &docker.HostConfig{
 			Binds:        BindsForTask(taskInfo),
 			PortBindings: PortBindingsForTask(taskInfo),
```